script-test/playbackstrategies/msestrategytest.js

```diff
@@ -1,491 +1,579 @@
+            expect(mseStrategy.isEnded()).toBe(false);
+          });
 
-          expect(mseStrategy.isPaused()).toBe(false);
-        });
+          it('should be set to true when we get an ended event', function () {
+            setUpMSE();
+            mseStrategy.load(null, null, 0);
 
-        it('should correctly return the paused state from the MediaPlayer when paused', function () {
-          setUpMSE();
-          mseStrategy.load(null, null, 0);
+            eventCallbacks('ended');
 
-          mockDashInstance.isPaused.and.returnValue(true);
+            expect(mseStrategy.isEnded()).toBe(true);
+          });
 
-          expect(mseStrategy.isPaused()).toBe(true);
-        });
-      });
+          it('should be set to false when we get a playing event', function () {
+            setUpMSE();
+            mseStrategy.load(null, null, 0);
 
-      describe('pause()', function () {
-        it('should call through to MediaPlayer\'s pause function', function () {
-          setUpMSE();
-          mseStrategy.load(null, null, 0);
+            eventCallbacks('playing');
 
-          mseStrategy.pause();
+            expect(mseStrategy.isEnded()).toBe(false);
+          });
 
-          expect(mockDashInstance.pause).toHaveBeenCalled();
-        });
-      });
+          it('should be set to false when we get a waiting event', function () {
+            setUpMSE();
+            mseStrategy.load(null, null, 0);
+
+            eventCallbacks('waiting');
+
+            expect(mseStrategy.isEnded()).toBe(false);
+          });
+
+          it('should be set to true when we get a completed event then false when we start initial buffering from playing', function () {
+            setUpMSE();
+            mseStrategy.load(null, null, 0);
 
-      describe('play()', function () {
-        it('should call through to MediaPlayer\'s play function', function () {
-          setUpMSE();
-          mseStrategy.load(null, null, 0);
+            eventCallbacks('ended');
 
-          mseStrategy.play();
+            expect(mseStrategy.isEnded()).toBe(true);
 
-          expect(mockDashInstance.play).toHaveBeenCalled();
+            eventCallbacks('waiting');
+
+            expect(mseStrategy.isEnded()).toBe(false);
+          });
         });
-      });
 
-      describe('setCurrentTime()', function () {
-        it('should call through to MediaPlayer\'s seek function', function () {
-          setUpMSE();
-          mseStrategy.load(null, null, 0);
+        describe('isPaused()', function () {
+          it('should correctly return the paused state from the MediaPlayer when not paused', function () {
+            setUpMSE();
+            mseStrategy.load(null, null, 0);
+
+            mockDashInstance.isPaused.and.returnValue(false);
+
+            expect(mseStrategy.isPaused()).toBe(false);
+          });
+
+          it('should correctly return the paused state from the audio MediaPlayer when not paused', function () {
```

Comment:
As per duration and seekable range above, I don't think we need to test paused for both audio and video.